src/decoder.rs

```diff
@@ -7,10 +7,14 @@ use {CborError, CborType};
 // any benign data.
 pub const MAX_ARRAY_SIZE: usize = 134_217_728;
 
+// Prevent stack exhaustion by limiting the nested depth of CBOR data.
+const MAX_NESTED_DEPTH: usize = 256;
```

Comment:
We should make sure this shows up somewhere in the docs. So we can either make this public like `MAX_ARRAY_SIZE` or put the number somewhere in a comment that gets into the docs.

Reply:
An alternative solution would be to try not to call `decode_item` all over again. Not sure if there's a nice way to do that.

Reply:
I added a comment to `decode`.
Unfortunately I can't think of a way to avoid this without completely re-working the implementation (and even then, I think it would be complicated to the point of introducing other bugs).